### PR TITLE
Exclude dummy backend in production

### PR DESCRIPTION
**Pull request by qwerty287**
folow-up https://github.com/woodpecker-ci/woodpecker/pull/3820

Really exclude the backend instead of replacing with a noop one.

### CODE REVIEW

File: docs/pnpm-lock.yaml [hunk 1]
```diff
@@ -14,22 +14,22 @@ importers:
     dependencies:
       '@docusaurus/core':
```
Comment (6543) [inline docs/pnpm-lock.yaml:15]: Unrelated

File: pipeline/backend/dummy/dummy.go [hunk 2]
```diff
@@ -12,9 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build test
```
Comment (6543) [inline pipeline/backend/dummy/dummy.go:15]: I would still let the tag stay

File: pipeline/backend/dummy/dummy_test.go [hunk 3]
```diff
@@ -12,9 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build test
```
Comment (6543) [inline pipeline/backend/dummy/dummy_test.go:15]: could you also re-add the tags here please?

Comment (anbraten) [reply pipeline/backend/dummy/dummy_test.go:15]: Same happens as mocks are excluded since #3831

**Review by 6543 — APPROVED** [review]
